Firestore/core/src/api/stages.h

```diff
@@ -194,6 +211,10 @@ class Where : public EvaluableStage {
     return "where";
   }
 
+  const Expr* expr() const {
```

Comment:
Any reason for using a raw pointer instead of a shared_ptr? We should generally avoid raw pointers whenever possible.